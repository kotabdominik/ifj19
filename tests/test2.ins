zadejte string a já zjistím jeho délku uwu
