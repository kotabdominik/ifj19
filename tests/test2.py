a = inputs()
b = len(a)
print('delka vlozeneho stringu', a, 'je', b)
