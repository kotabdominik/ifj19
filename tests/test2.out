.IFJcode19
JUMP $$MAIN
LABEL $ORD
CREATEFRAME
PUSHFRAME
DEFVAR LF@$RETVAL
DEFVAR LF@$POSIT
POPS LF@$POSIT
DEFVAR LF@$STRING
POPS LF@$STRING
DEFVAR LF@$CHR2INT
DEFVAR LF@$TMPLEN
STRLEN LF@$TMPLEN LF@$STRING
ADD LF@$TMPLEN LF@$TMPLEN int@-1
DEFVAR LF@$BOOLCHECK
GT LF@$BOOLCHECK LF@$POSIT LF@$TMPLEN
JUMPIFEQ $ERRORORD LF@$BOOLCHECK bool@true
LT LF@$BOOLCHECK LF@$POSIT int@0
JUMPIFEQ $ERRORORD LF@$BOOLCHECK bool@true
ADD LF@$TMPLEN LF@$TMPLEN int@1
STRI2INT LF@$RETVAL LF@$STRING LF@$POSIT
PUSHS LF@$RETVAL
POPFRAME
RETURN
LABEL $ERRORORD
MOVE LF@$RETVAL string@None
PUSHS LF@$RETVAL
POPFRAME
RETURN
LABEL $BETA
CREATEFRAME
PUSHFRAME
DEFVAR LF@$RETVAL
READ LF@$RETVAL int
PUSHS LF@$RETVAL
POPFRAME
RETURN
LABEL $SUBSTR
CREATEFRAME
PUSHFRAME
DEFVAR LF@$RETVAL
DEFVAR LF@$DELKA
POPS LF@$DELKA
DEFVAR LF@$POSIT
POPS LF@$POSIT
DEFVAR LF@$STRING
POPS LF@$STRING
DEFVAR LF@$TMPLEN
STRLEN LF@$TMPLEN LF@$STRING
DEFVAR LF@$BOOLCHECK
ADD LF@$TMPLEN LF@$TMPLEN int@-1
GT LF@$BOOLCHECK LF@$POSIT LF@$TMPLEN
JUMPIFEQ $ERRORSUB LF@$BOOLCHECK bool@true
LT LF@$BOOLCHECK LF@$POSIT int@0
JUMPIFEQ $ERRORSUB LF@$BOOLCHECK bool@true
GT LF@$BOOLCHECK int@0 LF@$DELKA
JUMPIFEQ $ERRORSUB LF@$BOOLCHECK bool@true
EQ LF@$BOOLCHECK int@0 LF@$DELKA
JUMPIFEQ $ERRORSUB LF@$BOOLCHECK bool@true
ADD LF@$TMPLEN LF@$TMPLEN int@1
GETCHAR LF@$RETVAL LF@$STRING LF@$POSIT
ADD LF@$POSIT LF@$POSIT int@1
DEFVAR LF@$TMPSTR
ADD LF@$DELKA LF@$DELKA LF@$POSIT
SUB LF@$DELKA LF@$DELKA int@1
LABEL $WHILE_LOOP_B
JUMPIFEQ $WHILE_LOOP_E LF@$POSIT LF@$DELKA
JUMPIFEQ $WHILE_LOOP_E LF@$POSIT LF@$TMPLEN
GETCHAR LF@$TMPSTR LF@$STRING LF@$POSIT
CONCAT LF@$RETVAL LF@$RETVAL LF@$TMPSTR
ADD LF@$POSIT LF@$POSIT int@1
JUMP $WHILE_LOOP_B
LABEL $WHILE_LOOP_E
PUSHS LF@$RETVAL
POPFRAME
RETURN
LABEL $ERRORSUB
MOVE LF@$RETVAL string@None
PUSHS LF@$RETVAL
POPFRAME
RETURN
LABEL $ALFA
CREATEFRAME
PUSHFRAME
DEFVAR LF@$RETVAL
READ LF@$RETVAL string
PUSHS LF@$RETVAL
POPFRAME
RETURN
LABEL $GAMA
CREATEFRAME
PUSHFRAME
DEFVAR LF@$RETVAL
READ LF@$RETVAL float
PUSHS LF@$RETVAL
POPFRAME
RETURN
LABEL $LENGTH
CREATEFRAME
PUSHFRAME
DEFVAR LF@$RETVAL
DEFVAR LF@$1
POPS LF@$1
STRLEN LF@$RETVAL LF@$1
PUSHS LF@$RETVAL
POPFRAME
RETURN
LABEL $CHAR
CREATEFRAME
PUSHFRAME
DEFVAR LF@$RETVAL
DEFVAR LF@$ASC
POPS LF@$ASC
INT2CHAR LF@$RETVAL LF@$ASC
PUSHS LF@$RETVAL
POPFRAME
RETURN
LABEL $ZEROERROR
EXIT int@9
LABEL $$MAIN
CREATEFRAME
PUSHFRAME
DEFVAR GF@$VARa
CALL $ALFA
POPS GF@$VARa
DEFVAR GF@$VARb
PUSHS GF@$VARa
CALL $LENGTH
POPS GF@$VARb
PUSHS string@delka\032vlozeneho\032stringu
PUSHS GF@$VARa
PUSHS string@je
PUSHS GF@$VARb
CREATEFRAME
PUSHFRAME
DEFVAR LF@$RETVAL0x7fffe75b3a54
DEFVAR LF@$PEPEGA0
POPS LF@$PEPEGA0
DEFVAR LF@$PEPEGA1
POPS LF@$PEPEGA1
DEFVAR LF@$PEPEGA2
POPS LF@$PEPEGA2
DEFVAR LF@$PEPEGA3
POPS LF@$PEPEGA3
WRITE LF@$PEPEGA3
WRITE string@\032
WRITE LF@$PEPEGA2
WRITE string@\032
WRITE LF@$PEPEGA1
WRITE string@\032
WRITE LF@$PEPEGA0
WRITE string@\010
MOVE LF@$RETVAL0x7fffe75b3a54  string@None
POPFRAME
